apphub/image_classification/cifar10_fast/cifar10_fast_tf.py

```diff
@@ -0,0 +1,134 @@
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Conv2D(num_channel, 3, padding='same')(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    return x
+
+
+def my_model():
+    #prep layers
+    inp = layers.Input(shape=(32, 32, 3))
+    x = layers.Conv2D(64, 3, padding='same')(inp)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    #layer1
+    x = layers.Conv2D(128, 3, padding='same')(x)
+    x = layers.MaxPool2D()(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Add()([x, residual(x, 128)])
+    #layer2
+    x = layers.Conv2D(256, 3, padding='same')(x)
+    x = layers.MaxPool2D()(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    #layer3
+    x = layers.Conv2D(512, 3, padding='same')(x)
+    x = layers.MaxPool2D()(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Add()([x, residual(x, 512)])
+    #layers4
+    x = layers.GlobalMaxPool2D()(x)
+    x = layers.Flatten()(x)
+    x = layers.Dense(10)(x)
+    x = layers.Activation('softmax', dtype='float32')(x)
+    model = tf.keras.Model(inputs=inp, outputs=x)
+
+    return model
+
+
+def get_estimator(epochs=24, batch_size=512, max_steps_per_epoch=None, save_dir=tempfile.mkdtemp()):
+    # step 1: prepare dataset
+    train_data, test_data = load_data()
+    pipeline = Pipeline(
+        train_data=train_data,
+        eval_data=test_data,
+        batch_size=batch_size,
+        ops=[
+            Normalize(inputs="x", outputs="x", mean=(0.4914, 0.4822, 0.4465), std=(0.2471, 0.2435, 0.2616)),
+            PadIfNeeded(min_height=40, min_width=40, image_in="x", image_out="x", mode="train"),
+            RandomCrop(32, 32, image_in="x", image_out="x", mode="train"),
+            Sometimes(HorizontalFlip(image_in="x", image_out="x", mode="train")),
+            CoarseDropout(inputs="x", outputs="x", mode="train", max_holes=1),
+            SmootOneHot(inputs="y", outputs="y", mode="train")
+        ])
+
+    # step 2: prepare network
+    model = fe.build(model_fn=my_model, optimizer_fn="adam")
+    network = fe.Network(ops=[
+        ModelOp(model=model, inputs="x", outputs="y_pred"),
+        CrossEntropy(inputs=("y_pred", "y"), outputs="ce"),
+        UpdateOp(model=model, loss_name="ce")
+    ])
+
+    # step 3: prepare estimator
+    estimator = fe.Estimator(
+        pipeline=pipeline,
+        network=network,
+        epochs=epochs,
+        traces=[
+            Accuracy(true_key="y", pred_key="y_pred"),
+            BestModelSaver(model=model, save_dir=save_dir, metric="accuracy", save_best_mode="max")
+        ],
+        max_steps_per_epoch=max_steps_per_epoch)
+
+    return estimator
+
+
+if __name__ == "__main__":
+    est = get_estimator()
+    est.fit()
```

Comment:
@vbvg2008 has been putting .test() calls everywhere too. do we want that to be standard across all app hubs?

Reply:
in this case, yes.   but the test won't go to every case